### PR TITLE
Added support for new dateTextContent attr in GlobalRule

### PR DESCRIPTION
**Pull request by mburak**
This PR

* adds support for a new attribute dateTextContent used in GlobalRule for dates that are in the content side and not in the datetime attribute.

### CODE REVIEW

**Comment by diegoquinteiro** [comment]
I don't clearly understand what this PR does. Can you give more context of how the rules will look like?

File: src/Facebook/InstantArticles/Transformer/Getters/DateGetter.php [hunk 1]
```diff
@@ -50,10 +50,10 @@ public function get($node)
         if (!empty($elements) && $elements->item(0)) {
             $element = $elements->item(0);
 
-            if ($this->attribute) {
+            if ($this->attribute && $this->attribute != 'dateTextContent') {
                 return \DateTime::createFromFormat($this->format, $element->getAttribute($this->attribute));
             }
-            return \DateTime::createFromFormat($this->format, $element->textContent);
+            return \DateTime::createFromFormat('!'.$this->format, trim($element->textContent));
```
Comment (diegoquinteiro) [inline src/Facebook/InstantArticles/Transformer/Getters/DateGetter.php]: Maybe that's the only line you should keep (plus the tests).

File: src/Facebook/InstantArticles/Transformer/Getters/DateGetter.php [hunk 2]
```diff
@@ -50,10 +50,10 @@ public function get($node)
         if (!empty($elements) && $elements->item(0)) {
             $element = $elements->item(0);
 
-            if ($this->attribute) {
+            if ($this->attribute && $this->attribute != 'dateTextContent') {
```
Comment (diegoquinteiro) [inline src/Facebook/InstantArticles/Transformer/Getters/DateGetter.php]: Let's not use a custom attribute name like 'dateTextContent'. We can simply omit the attribute on the rule to fetch the content, that's the default for all rules and getters.

File: tests/Facebook/InstantArticles/Transformer/GlobalHtml/global-rules-with-text-date.json [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+                },
+                "article.canonical": {
+                    "type" : "string",
+                    "selector": "head meta",
+                    "attribute": "content"
+                },
+                "article.publish": {
+                    "selector": "div.date",
+                    "type": "date",
+                    "attribute": "dateTextContent",
```
Comment (diegoquinteiro) [inline tests/Facebook/InstantArticles/Transformer/GlobalHtml/global-rules-with-text-date.json]: Just remove this line to accomplish the same behavior.

File: tests/Facebook/InstantArticles/Transformer/TransformerTest.php [hunk 4]
```diff
@@ -92,8 +92,6 @@ public function testSelfTransformerContent()
         $instant_article->addMetaProperty('op:generator:transformer:version', '1.0.0');
         $result = $instant_article->render('', true)."\n";
 
-        //var_dump($result);
```
Comment (diegoquinteiro) [inline tests/Facebook/InstantArticles/Transformer/TransformerTest.php]: 💅 👍

**Comment by mburak** [comment]
fixes based on @diegoquinteiro comments

**Pull request merged**